### DISCUSSION TITLE
Getting MIDI working on TR1 - RC_0.0N4

### DISCUSSION THREAD

**Discussion by ToniB197** [eh2k/squares-and-circles#33] [Q&A]
Hi, this is my 1st post on GitHub so I hope my post is in alignment with the rules :-)
I just got the uO_C squares-and-circles RC_0.0N4.
I'm trying to get a MIDI signal in to T1 to trigger a drum e.g. 909ish-BD
I'm sending a note C4 (60) on CH1 (USB type A) to another MIDI device and it works fine, but when I plug that MIDI cable into T1 nothing happens.
Did I overlook or configure something wrong?
Any help is highly appreciated
Best, Toni
(See some settings attached)
![uO_C squares-and-circles release RC_0 0N4](https://user-images.githubusercontent.com/123867504/215323951-0f867b30-4f6c-4d36-b533-8e0fcc2a0d10.jpg)

**Comment by eh2k** [comment]
Hi @ToniB197 

sounds like the same issue #31.

I just tried this again on my device - and it works. You have actually done everything right so far:
 * `Engine #1` is assigned to midi channel 1.
 * The midi monitor is running on `#1`.

The only thing you could try is to change the midi input from T1 to USB and back again (there were reports that it somehow doesn't always re-initialize after rebooting #21).

Otherwise, I suspect that somehow the connection does not fit electrically. Can you tell me exactly which adapter you are using.

**Reply by ToniB197** [reply]
Hi @eh2k 
Thanks for getting back to me.
Sorry I overlooked the issues page, guess I posted wrong after all :-)
I tried to change from T1 to USB and back and vice versa + saving but it didn't work.
I'm using a straight TRS mini jack cable (MIDI signal works fine to another device MIDI type A) 
Should I not be able to see something on the page MIDI/Monitor if all works as intended?

**Reply by eh2k** [reply]
All good - it doesn't matter if it's an issue or a discussion - imho it's better to have similar issues than mixed topics.

I'm sorry to say that the midi input on TR1 will not work with any of the usual midi adapters. As far as I understand not even with a Type-C TS adapter. For it to work, ground must be connected to the sleeve, and a 0V/5V level must be applied to the tip, so that the Q1 switches, and the Teensy can read it as a serial data stream on the GPIO pin. I built a small 1U module for this, this is the schematic for it. Besides the OC I have a few other modules connected to it that have headers on the PCB for Midi-In. The lower right circuit with the Q1 is on the OC. <img src="https://github.com/eh2k/squares-and-circles/raw/main/doc/midi2ts.png" />

**Reply by ToniB197** [reply]
you are not selling such a 1U module by any chance? :-)
I do know how to solder but am not experience enough to translate the schematic into a module.

**Reply by eh2k** [reply]
sorry - I built mine on a simple breadboard.
Maybe this will help you - nice project to get into diy:-) 

https://oshwlab.com/yarogniew/MIDI_in_konektor-cfad43e472c940898cfe784b4477587a

**Comment by ericmodular** [comment]
I have a really dumb question. What is the advantage of using TR1 for MIDI input instead of the USB port on the Teensy? Aside from having to run a cable to the back of the module.

**Comment by eh2k** [comment]
Hi @ericmodular 

I don't have any music devices that support midi via USB - they all still have the old-school midi interfaces. The TR1 (by chance) offered itself as a possible midi input (with the above-mentioned circuit). Functionally it doesn't matter. As USB is generally a more complex interface, it's not easy to test. I hope everything works as theoretically assumed.

**Reply by ericmodular** [reply]
Thanks for the response. I don’t know why I didn’t think about traditional MIDI devices but of course that makes sense. I’m going to try building that adapter circuit, making it into a little module is a good idea since there is a 5V power rail available.
